src/users/__tests__/userHelpers-test.js

```diff
@@ -271,30 +368,75 @@ describe('groupUsersByStatus', () => {
+    expect(groupedUsers).toMatchObject({ active: [], idle: [], unavailable: [], offline: [] });
   });
 
   test('sort input by status, when no presence entry consider offline', () => {
     const users = deepFreeze([
-      { email: 'allen@example.com' },
-      { email: 'bob@example.com' },
-      { email: 'carter@example.com' },
-      { email: 'dan@example.com' },
+      { full_name: 'Mobile User', email: 'allen@example.com' },
+      { full_name: 'Example User', email: 'bob@example.com' },
+      { full_name: 'Carter User1', email: 'carter@example.com' },
+      { full_name: 'Zulip User', email: 'dan@example.com' },
     ]);
-    const presence = {
-      'allen@example.com': { aggregated: { status: 'active' } },
-      'bob@example.com': { aggregated: { status: 'idle', timestamp: Date.now() / 1000 - 10 } },
-      'carter@example.com': { aggregated: { status: 'offline' } },
+
+    const presences = {
+      'allen@example.com': {
+        aggregated: { status: 'active', timestamp: Date.now() / 1000, client: 'ZulipMobile' },
+        ZulipMobile: {
+          status: 'active',
+          timestamp: Date.now() / 1000,
+          client: 'ZulipMobile',
+          pushable: false,
+        },
+      },
+      'bob@example.com': {
+        aggregated: { status: 'idle', timestamp: Date.now() / 1000 - 10, client: 'website' },
+        website: {
+          status: 'idle',
+          timestamp: Date.now() / 1000 - 10,
+          client: 'website',
+          pushable: true,
+        },
+      },
+      'carter@example.com': {
+        aggregated: { status: 'offline', timestamp: Date.now() / 10000, client: 'ZulipMobile' },
+        ZulipMobile: {
+          status: 'offline',
+          timestamp: Date.now() / 10000,
+          client: 'ZulipMobile',
+          pushable: false,
+        },
+      },
     };
+
     const expectedResult = {
-      active: [{ email: 'allen@example.com' }],
-      idle: [{ email: 'bob@example.com' }],
-      offline: [{ email: 'carter@example.com' }, { email: 'dan@example.com' }],
+      active: [{ full_name: 'Mobile User', email: 'allen@example.com' }],
+      idle: [{ full_name: 'Example User', email: 'bob@example.com' }],
+      offline: [
+        { full_name: 'Carter User1', email: 'carter@example.com' },
+        { full_name: 'Zulip User', email: 'dan@example.com' },
+      ],
       unavailable: [],
     };
 
-    const groupedUsers = groupUsersByStatus(users, presence);
-    expect(groupedUsers).toEqual(expectedResult);
+    const modifiedPresences = {};
+    Object.keys(presences).forEach(user => {
+      const userStatus = {};
+      const email: string = user;
+      const userPresence = presences[email];
+      Object.keys(userPresence).forEach(key => {
+        const presenceKey: string = key;
+        userStatus[presenceKey] = userPresence[presenceKey];
+      });
+      modifiedPresences[email] = userStatus;
+    });
```

Comment:
If this were needed, it would definitely need to be encapsulated in its own function, rather than used inline twice.